test/extended/util/framework.go

```diff
@@ -318,12 +398,27 @@ func WaitForADeployment(client kclient.ReplicationControllerInterface, name stri
 		if err != nil {
 			return err, false
 		}
+		// multiple deployments are conceivable; so we look to see how the latest depoy does
+		lastMatched := false
+		var lastErr error = nil
+		var lastRC string
 		for _, rc := range list.Items {
-			err, matched := okOrFailed(&rc)
-			if matched {
-				return err, false
+			if lastRC <= rc.GetName() {
```

Comment:
i'm guessing we're not too worried about it, but this will fail if we get beyond 9 deployments.